### PR TITLE
support SM2 signature and SM3 hash

### PR DESCRIPTION
**Pull request by zhihuixinyuan**
添加sm2算法

### CODE REVIEW

File: packages/cita-signer/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+//两种签名算法: sm2  secp256k1
```
Comment (Keith-CY) [inline packages/cita-signer/src/config.ts]: Please add comment in English

File: packages/cita-signer/package.json [hunk 2]
```diff
@@ -16,9 +16,11 @@
     "test:v2": "cross-env VERSION=2 ../../node_modules/.bin/jest"
   },
   "dependencies": {
+    "ec": "^0.0.1",
     "elliptic": "^6.4.0",
     "fstream": "1.0.12",
     "google-protobuf": "^3.5.0",
+    "sm-series-crypto": "^0.2.0",
```
Comment (Keith-CY) [inline packages/cita-signer/package.json]: @kaikai1024 Please evaluate the reliability of this package

File: packages/cita-signer/src/index.ts [hunk 3]
```diff
@@ -173,32 +177,81 @@ const signer = (
...
+    const sign_buffer = Buffer.from(signature, 'hex')
+    const sigBytes = new Uint8Array(65)
+    sigBytes.set(sign_buffer)
+
+    const unverifiedTx = new blockchainPb.UnverifiedTransaction()
+    unverifiedTx.setTransaction(tx)
+    const serializedUnverifiedTx = unverifiedTx.serializeBinary()
+    var hexUnverifiedTx = exports.bytes2hex(serializedUnverifiedTx)
+    hexUnverifiedTx =
+      hexUnverifiedTx + '128001' + signature + publicKey.slice(2)
```
Comment (Keith-CY) [inline packages/cita-signer/src/index.ts]: Don't use the magic number, please name the constant of `128001`

File: packages/cita-signer/package.json [hunk 4]
```diff
@@ -16,9 +16,11 @@
     "test:v2": "cross-env VERSION=2 ../../node_modules/.bin/jest"
   },
   "dependencies": {
+    "ec": "^0.0.1",
```
Comment (Keith-CY) [inline packages/cita-signer/package.json]: It seems that this package is not used.

Comment (kaikai1024) [reply packages/cita-signer/package.json]: Yes. It has license problem.

File: packages/cita-signer/package.json (hunk 2 above)
Comment (kaikai1024) [reply packages/cita-signer/package.json]: I'm not sure. The license is ok.

**Pull request merged**